File: examples/javascript/vector-store.ts
import type { EmbeddingsInterface } from "@langchain/core/embeddings";
import type { Document } from "@langchain/core/documents";
import type { Collection } from "mongodb";
import {
  DocumentDBVectorStore,
  type DocumentDBStoredDocument,
} from "@documentdb/langchain-documentdb";

export async function run(
  collection: Collection<DocumentDBStoredDocument>,
  embeddings: EmbeddingsInterface,
): Promise<Document[]> {
  const store = new DocumentDBVectorStore(embeddings, {
    collection,
    dimensions: 1536,
    indexName: "embedding_vector",
    indexKind: "diskann",
    similarity: "COS",
    diskAnnMaxDegree: 32,
    diskAnnBuildCandidates: 50,
    diskAnnSearchCandidates: 80,
  });

  await store.ensureCollectionExists();
  await store.addDocuments([
    {
      id: "documentdb-overview",
      pageContent: "DocumentDB provides a MongoDB-compatible API.",
      metadata: { category: "database", published: true },
    },
    {
      id: "langchain-overview",
      pageContent: "LangChain composes retrieval with language models.",
      metadata: { category: "framework", published: true },
    },
  ]);

  return store.similaritySearch(
    "Which database works with MongoDB drivers?",
    4,
    {
      "metadata.category": "database",
    },
  );
}

File: examples/go/vector_store.go
package examples

import (
	"context"

	"github.com/documentdb/langchain-documentdb/go/documentdb"
	"github.com/tmc/langchaingo/embeddings"
	"github.com/tmc/langchaingo/schema"
	"github.com/tmc/langchaingo/vectorstores"
	"go.mongodb.org/mongo-driver/v2/bson"
	"go.mongodb.org/mongo-driver/v2/mongo"
)

func Run(ctx context.Context, database *mongo.Database, embedder embeddings.Embedder) ([]schema.Document, error) {
	store, err := documentdb.New(documentdb.Config{
		Database:                database,
		CollectionName:          "documents",
		Embedder:                embedder,
		Dimensions:              1536,
		IndexKind:               documentdb.IndexDiskANN,
		Similarity:              documentdb.SimilarityCosine,
		DiskANNMaxDegree:        32,
		DiskANNBuildCandidates:  50,
		DiskANNSearchCandidates: 80,
	})
	if err != nil {
		return nil, err
	}
	if err := store.EnsureCollectionExists(ctx); err != nil {
		return nil, err
	}
	_, err = store.AddDocuments(ctx, []schema.Document{
		{PageContent: "DocumentDB provides a MongoDB-compatible API.", Metadata: map[string]any{"id": "documentdb-overview", "category": "database"}},
		{PageContent: "LangChain composes retrieval with language models.", Metadata: map[string]any{"id": "langchain-overview", "category": "framework"}},
	})
	if err != nil {
		return nil, err
	}
	return store.SimilaritySearch(
		ctx,
		"Which database works with MongoDB drivers?",
		4,
		vectorstores.WithFilters(bson.D{{Key: "category", Value: "database"}}),
		vectorstores.WithScoreThreshold(0.75),
	)
}
